samples/layouts/dock-manager/add-content-runtime/src/index.ts

```diff
@@ -0,0 +1,233 @@
+                    orientation: IgcSplitPaneOrientation.horizontal,
+                    floatingHeight: 150,
+                    floatingWidth: 250,
+                    floatingLocation: { x: 300, y: 200 },
+                    panes: [
+                        {
+                            type: IgcDockManagerPaneType.contentPane,
+                            contentId: 'content9',
+                            header: 'Floating Pane'
+                        },
+                    ]
+                },
+            ]
+        };
+
+        addContentPaneBtn!.addEventListener('click', () => {
+            const cp: IgcContentPane = {
+              type: IgcDockManagerPaneType.contentPane,
+              contentId: `newContent${this._counter}`,
+              header: `New Pane ${this._counter}`,
+            };
+
+            this.dockManager.layout.rootPane.panes.push(cp);
+        
+            this.attachPane();
+        })
+
+        addTabGroupPaneBtn!.addEventListener('click', () => {
+            const cp: IgcContentPane = {
+              type: IgcDockManagerPaneType.contentPane,
+              contentId: `newContent${this._counter}`,
+              header: `New Document ${this._counter}`,
+            };
+
+            const splitPanes = this.dockManager.layout.rootPane.panes.filter(p => p.type === 'splitPane') as IgcSplitPane[];
+            let parentDocumentHost = splitPanes.find(sp => sp.panes.some(spp => spp.type === 'documentHost'));
+
+            if (parentDocumentHost === undefined) {
+                parentDocumentHost = {
+                    type: IgcDockManagerPaneType.splitPane,
+                    orientation: IgcSplitPaneOrientation.vertical,
+                    size: 200,
+                    panes: [
+                        {
+                            type: IgcDockManagerPaneType.documentHost,
+                            size: 200,
+                            rootPane: {
+                                type: IgcDockManagerPaneType.splitPane,
+                                orientation: IgcSplitPaneOrientation.horizontal,
+                                panes: []
+                            }
+                        }
+                    ]
+                }
+
+                this.dockManager.layout.rootPane.panes.push(parentDocumentHost);
+            }
+
+            const childDocumentHost = parentDocumentHost!.panes.find(p => p.type === 'documentHost') as IgcDocumentHost;
+            let tabGroup = childDocumentHost.rootPane.panes[0];
+    
+            if (tabGroup) {
+                if (tabGroup.type !== 'tabGroupPane') {
+                    tabGroup = (tabGroup as IgcSplitPane).panes.find(p => p.type === 'tabGroupPane') as IgcTabGroupPane;
+                }
+
+                tabGroup.panes.push(cp);
+            } else {
+                const contentPanes = childDocumentHost.rootPane.panes.filter(p => p.type === 'contentPane') as IgcContentPane[];
```

Comment:
If we end up here, doesn't this mean the rootPane.panes collection is empty? If so, why searching for content panes?